### PR TITLE
Api Endpoint-Get user by Role created

### PR DESCRIPTION
**Pull request by fawadik**
Api endpoint created for getting users by role

### CODE REVIEW

File: src/server/api/routes/user.router.js [hunk 1]
```diff
@@ -23,6 +23,18 @@ router.get("/", authorizeUser(ROLES.SUPER_USER), (req, res, next) => {
     .catch(next);
 });
 
+// ENDPOINT: /api/users/:role :GET
```
Comment (ismailirahim) [inline src/server/api/routes/user.router.js]: Fowad, i was talking with @zkwsk about the endpoint and that should be /api/user-roles/:user_role_id/users

**Review by ismailirahim — CHANGES_REQUESTED** [review]
change the endpoint to /api/user-roles/:user_role_id/users

File: src/server/api/routes/user.router.js (hunk 1 above)
Comment (fawadik) [reply src/server/api/routes/user.router.js]: rahim... user_role_id or only role?
bcoz zaki said to me that instead of id it should be on role

Comment (zkwsk) [reply src/server/api/routes/user.router.js]: I'm sorry if I said something wrong. I see that it was also not described in the Trello at all. But Rahim is right about the path.

Comment (fawadik) [reply src/server/api/routes/user.router.js]: I have created the new router and now modified the right path.
please have a look.

**Review by Swapna-Sahu — DISMISSED** [review]
Looks good to me

File: src/server/api/controllers/user.controller.js [hunk 2]
```diff
@@ -33,6 +33,21 @@ const getUserById = async id => {
   }
 };
 
+const getUsersByRole = async role => {
+  try {
+    const users = await knex("users")
+      .join("user_roles", "user_roles.id", "users.role_id")
+      .select("*")
+      .where({ role: role });
```
Comment (zkwsk) [inline src/server/api/controllers/user.controller.js:41]: You need to query by the `role_id`. Currently it does not work.

```suggestion
      .where({ role_id: role });
```

File: src/server/api/routes/user-roles.router.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const { authorizeUser } = require("../lib/middleware/auth.middleware");
+
+const ROLES = require("../../constants/roles");
+
+// controllers
+const userController = require("../controllers/user.controller");
+
+// ENDPOINT: /api/user-roles/:role/users :GET
+router.get(
+  "/:role/users",
```
Comment (zkwsk) [inline src/server/api/routes/user-roles.router.js:18]: Can you please add documentation for this endpoint to Swagger?

**Review by zkwsk — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/user.controller.js (hunk 2 above)
Comment (zkwsk) [reply src/server/api/controllers/user.controller.js:41]: Sorry I was using the wrong request body. This works with role name (i.e. "admin")

**Review by zkwsk — APPROVED** [review]

**Pull request merged**